app/api/tax.py

```diff
@@ -103,17 +92,26 @@ def before_get(self, args, kwargs):
         :param kwargs:
         :return:
         """
-        tax = Tax.query.filter_by(id=kwargs['id']).one()
-        if 'Authorization' in request.headers and has_access('is_coorganizer', event_id=tax.event_id):
-            self.schema = TaxSchema
+        if 'id' in kwargs:
+            tax = Tax.query.filter_by(id=kwargs['id']).one()
```

Comment:
I think when using `.one()` we also check NoResultFound exception in this project ex: https://github.com/fossasia/open-event-server/blob/development/app/api/events.py#LC294
Reference
https://stackoverflow.com/questions/18110033/getting-first-row-from-sqlalchemy#answer-18110236

Reply:
What if we use `.first()` instead of `.one()`?

Reply:
Yeah, I will use `.first()` and then do a null check. We can also use `.one()` and except a NoResultFound exception. Both will work.

Reply:
When there is need to pop out message to user then using .one() is best (as we have exceptions which point to which attribute or relationship is incorrectly added by user ) otherwise .first() will not deal these.

Yes, But .one() is preferred in IMO due to poping out the the reason for the exception. It has been used much of the times here at oe-server.

Reply:
We can raise an error with `.first()` also. 
``` Python
tax = Tax.query.filter_by(id=kwargs['id']).first()
if not tax:
   raise error here
```

Reply:
Yes, but if we use ResultNotFound and .one() we can easily say we have sticked to one way in this project of dealing such situations. IMO.